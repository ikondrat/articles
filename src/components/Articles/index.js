import React from 'react';
import Divider from '@material-ui/core/Divider'
import { PropTypes } from 'prop-types'
import {
    Link
} from 'react-router-dom'
import {
    List,
    ListItem
} from '@material-ui/core'
import {
    Progress,
    Pager,
    Bar
} from '../../components';

const Articles = ({fetchArticles, fetchMore, list, isFetching, pages, page, query}) => (
    <div>
        <Bar onChange={(v) => fetchArticles({query: v})}/>
        <Divider/>
        {list && list.length > 0 &&
            <List component="nav">
            {list.map(({objectID, title}, i) => (
                <ListItem button key={i}>
                <Link to={`/articles/${objectID}`}>
                    {title}
                </Link>
                </ListItem>
            ))}
            </List>
        }
        {isFetching &&
            <Progress/>
        }
        <Divider/>
        {pages > page + 1 &&
            <Pager 
                pages={pages}
                onClickMore={(p) => fetchMore({page: page, query: query})}
                page={page}
            />
        }
    </div>
)

Articles.propTypes = {
    onChange: PropTypes.func,
    list: PropTypes.array,
    isFetching: PropTypes.bool,
    pages: PropTypes.number,
    page: PropTypes.number,
    fetchArticles: PropTypes.func,
};

export default Articles;