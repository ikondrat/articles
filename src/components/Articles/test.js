import React from 'react';
import { shallow } from 'enzyme';
import Articles from './';

it('renders correctly', () => {
  const wrapper = shallow(
    <Articles 
      title="test title"
      isFetching={true}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly with list', () => {
  const mockfetchArticles = jest.fn();
  const mockfetchMore = jest.fn();
  const wrapper = shallow(
    <Articles 
      title="test title"
      isFetching={false}
      page={0}
      pages={10}
      fetchArticles={mockfetchArticles}
      fetchMore={mockfetchMore}
      list={[
        {
          title: "Title1",
          objectID: "123"
        },
        {
          title: "Title2",
          objectID: "124"
        }
      ]}
    />
  );
  expect(wrapper).toMatchSnapshot();
});