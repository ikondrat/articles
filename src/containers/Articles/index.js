import React, { Component } from 'react'
import { connect } from 'react-redux'
import {
  fetchArticles,
  fetchMore,
} from '../../actions'
import Articles from '../../components/Articles'

class ArticlesContainer extends Component {
  componentWillMount() {
    this.props.fetchArticles();
  }
  render() {
    return (<Articles {...this.props}/>)
  }
}

const bindActionsToDispatch = dispatch => ({
    fetchArticles : (x) => {dispatch(fetchArticles(x))},
    fetchMore: (params) => {dispatch(fetchMore(params))}
});

const mapStateToProps = (s) => {
  return s.articles;
}

export default connect(mapStateToProps, bindActionsToDispatch)(ArticlesContainer)