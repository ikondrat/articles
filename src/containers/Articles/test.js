import React from 'react';
import { shallow } from 'enzyme';
import Articles from '.';
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureMockStore();
const store = mockStore({
  articles: {
    isFetching: true,
    list: [
      {
        title: 'Hello'
      },
      {
        title: 'Ooops'
      }
    ]
  }
});

it('renders correctly', () => {
  const wrapper = shallow(
    <Provider store={store}>
      <Articles />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});