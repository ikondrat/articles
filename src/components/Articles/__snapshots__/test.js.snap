// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Articles
    isFetching={true}
    title="test title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Bar
          onChange={[Function]}
/>,
        <WithStyles(Divider) />,
        undefined,
        <Unknown />,
        <WithStyles(Divider) />,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Bar
            onChange={[Function]}
/>,
          <WithStyles(Divider) />,
          undefined,
          <Unknown />,
          <WithStyles(Divider) />,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`renders correctly with list 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Articles
    fetchArticles={[Function]}
    fetchMore={[Function]}
    isFetching={false}
    list={
        Array [
            Object {
              "objectID": "123",
              "title": "Title1",
            },
            Object {
              "objectID": "124",
              "title": "Title2",
            },
          ]
    }
    page={0}
    pages={10}
    title="test title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Bar
          onChange={[Function]}
/>,
        <WithStyles(Divider) />,
        <WithStyles(List)
          component="nav"
>
          <WithStyles(ListItem)
                    button={true}
          >
                    <Link
                              replace={false}
                              to="/articles/123"
                    >
                              Title1
                    </Link>
          </WithStyles(ListItem)>
          <WithStyles(ListItem)
                    button={true}
          >
                    <Link
                              replace={false}
                              to="/articles/124"
                    >
                              Title2
                    </Link>
          </WithStyles(ListItem)>
</WithStyles(List)>,
        false,
        <WithStyles(Divider) />,
        <Pager
          onClickMore={[Function]}
          page={0}
          pages={10}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(ListItem)
              button={true}
>
              <Link
                            replace={false}
                            to="/articles/123"
              >
                            Title1
              </Link>
</WithStyles(ListItem)>,
            <WithStyles(ListItem)
              button={true}
>
              <Link
                            replace={false}
                            to="/articles/124"
              >
                            Title2
              </Link>
</WithStyles(ListItem)>,
          ],
          "component": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "button": true,
              "children": <Link
                replace={false}
                to="/articles/123"
>
                Title1
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Title1",
                "replace": false,
                "to": "/articles/123",
              },
              "ref": null,
              "rendered": "Title1",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "button": true,
              "children": <Link
                replace={false}
                to="/articles/124"
>
                Title2
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Title2",
                "replace": false,
                "to": "/articles/124",
              },
              "ref": null,
              "rendered": "Title2",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onClickMore": [Function],
          "page": 0,
          "pages": 10,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Bar
            onChange={[Function]}
/>,
          <WithStyles(Divider) />,
          <WithStyles(List)
            component="nav"
>
            <WithStyles(ListItem)
                        button={true}
            >
                        <Link
                                    replace={false}
                                    to="/articles/123"
                        >
                                    Title1
                        </Link>
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
                        button={true}
            >
                        <Link
                                    replace={false}
                                    to="/articles/124"
                        >
                                    Title2
                        </Link>
            </WithStyles(ListItem)>
</WithStyles(List)>,
          false,
          <WithStyles(Divider) />,
          <Pager
            onClickMore={[Function]}
            page={0}
            pages={10}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(ListItem)
                button={true}
>
                <Link
                                replace={false}
                                to="/articles/123"
                >
                                Title1
                </Link>
</WithStyles(ListItem)>,
              <WithStyles(ListItem)
                button={true}
>
                <Link
                                replace={false}
                                to="/articles/124"
                >
                                Title2
                </Link>
</WithStyles(ListItem)>,
            ],
            "component": "nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "button": true,
                "children": <Link
                  replace={false}
                  to="/articles/123"
>
                  Title1
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Title1",
                  "replace": false,
                  "to": "/articles/123",
                },
                "ref": null,
                "rendered": "Title1",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "button": true,
                "children": <Link
                  replace={false}
                  to="/articles/124"
>
                  Title2
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Title2",
                  "replace": false,
                  "to": "/articles/124",
                },
                "ref": null,
                "rendered": "Title2",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onClickMore": [Function],
            "page": 0,
            "pages": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
